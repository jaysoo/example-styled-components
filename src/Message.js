import React from 'react'
import styled from 'styled-components';

const Message = styled(({ children, className }) => (
  <p className={className}>
    {children}
  </p>
))`
  font-size: 24px;
  color: ${props => props.color || 'palevioletred'};
`;

export default Message;
