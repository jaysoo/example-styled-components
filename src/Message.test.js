import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Message from './Message';

it('renders', () => {
  const component = renderer.create(
    <Message>
      Hello!
    </Message>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});
