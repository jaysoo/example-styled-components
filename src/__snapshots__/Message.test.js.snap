// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.c0 {
  font-size: 24px;
  color: palevioletred;
}

<p
  className="c0"
>
  Hello!
</p>
`;
